src/entities/repos/submission_repo.py

```diff
@@ -140,6 +147,39 @@ async def update_task_state(
     await upsert_helper(session, task, FilingTaskProgressDAO)
 
 
+async def update_contact_info(
+    session: AsyncSession, lei: str, filing_period: str, new_contact_info: ContactInfoDTO
+) -> ContactInfoDAO:
+    filing = await get_filing(session, lei=lei, filing_period=filing_period)
+    found_contact_info = await query_helper(session, ContactInfoDAO, filing=filing.id)
+    if found_contact_info:
+        contact_info = found_contact_info[0]
+        contact_info = filing.contact_info
+        contact_info.first_name = new_contact_info.first_name
+        contact_info.last_name = new_contact_info.last_name
+        contact_info.hq_address_street_1 = new_contact_info.hq_address_street_1
+        contact_info.hq_address_street_2 = new_contact_info.hq_address_street_2
+        contact_info.hq_address_city = new_contact_info.hq_address_city
+        contact_info.hq_address_state = new_contact_info.hq_address_state
+        contact_info.hq_address_zip = new_contact_info.hq_address_zip
+        contact_info.email = new_contact_info.email
+        contact_info.phone = new_contact_info.phone
+    else:
+        contact_info = ContactInfoDAO(
+            filing=filing.id,
+            first_name=new_contact_info.first_name,
+            last_name=new_contact_info.last_name,
+            hq_address_street_1=new_contact_info.hq_address_street_1,
+            hq_address_street_2=new_contact_info.hq_address_street_2,
+            hq_address_city=new_contact_info.hq_address_city,
+            hq_address_state=new_contact_info.hq_address_state,
+            hq_address_zip=new_contact_info.hq_address_zip,
+            email=new_contact_info.email,
+            phone=new_contact_info.phone,
+        )
```

Comment:
since dto and dao should have the same fields, I believe the `upsert_helper` will take care of all the field setting for you; so I think just calling `upsert_helper(..., new_contact_info, ...)` should suffice?

Reply:
actually... still need filing id, so probably something like `ContactInfoDAO(**new_contact_info.__dict__.copy(), filing=filing.id)`

Reply:
Checking now.